Fuzzers fail to build without jxl_extras-static

Hello,

I am experimenting with including `libjxl` into Android application. I am building `libjx` on Windows in Android Studio and the app builds via Gradle.

I observed that **by default**, all cmake targets are built there (unless I manually specify the exact target I want).

I was getting following errors:

`C:\Devel\Camera\PhotonCamera\app\src\main\cpp\deps\libjxl\tools\fuzzer_stub.cc:11:10: fatal error: 'jxl/thread_parallel_runner.h' file not found`

`C:\Devel\Camera\PhotonCamera\app\src\main\cpp\deps\libjxl\tools\cjxl_fuzzer.cc:18:10: fatal error: 'jxl/encode.h' file not found`

The problem is, that majority of fuzzer runners fail to build when **JPEGXL_ENABLE_TOOLS** is not enabled, because they need **jxl_extras-static** target, which depends on that option:

```
if(JPEGXL_ENABLE_TOOLS)
  include(jxl_extras.cmake)
endif()
```

I am able to reproduce the build problem under Linux too, when attempting to build `_fuzzer_runner` target explicitly:
```
make fields_fuzzer_runner
[  0%] Built target tool_version_git
[ 40%] Built target hwy
[ 80%] Built target jxl_tool
[ 80%] Building CXX object tools/CMakeFiles/fields_fuzzer_runner.dir/fuzzer_stub.cc.o
[ 80%] Building CXX object tools/CMakeFiles/fields_fuzzer_runner.dir/fields_fuzzer.cc.o
[100%] Linking CXX executable fields_fuzzer_runner
/usr/lib/gcc/x86_64-pc-linux-gnu/12/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -ljxl_extras-static: Adresár alebo súbor neexistuje
collect2: error: ld returned 1 exit status
make[3]: *** [tools/CMakeFiles/fields_fuzzer_runner.dir/build.make:124: tools/fields_fuzzer_runner] Error 1
make[2]: *** [CMakeFiles/Makefile2:873: tools/CMakeFiles/fields_fuzzer_runner.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:880: tools/CMakeFiles/fields_fuzzer_runner.dir/rule] Error 2
make: *** [Makefile:478: fields_fuzzer_runner] Error 2
```

```
make color_encoding_fuzzer_runner
[  0%] Built target tool_version_git
[ 40%] Built target hwy
[ 80%] Built target jxl_tool
[ 80%] Linking CXX executable color_encoding_fuzzer_runner
/usr/lib/gcc/x86_64-pc-linux-gnu/12/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -ljxl_extras-static: Adresár alebo súbor neexistuje
collect2: error: ld returned 1 exit status
make[3]: *** [tools/CMakeFiles/color_encoding_fuzzer_runner.dir/build.make:124: tools/color_encoding_fuzzer_runner] Error 1
make[2]: *** [CMakeFiles/Makefile2:735: tools/CMakeFiles/color_encoding_fuzzer_runner.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:742: tools/CMakeFiles/color_encoding_fuzzer_runner.dir/rule] Error 2
make: *** [Makefile:413: color_encoding_fuzzer_runner] Error 2
```

Only `djxl_fuzzer_runner` builds because it used different target_link_libraries.

I suggest that instead of https://github.com/libjxl/libjxl/blob/9b367dcf5f01d72b7d0159597613e789e10036e3/tools/CMakeLists.txt#L310

to use something like:
```
set(FUZZER_BINARIES djxl_fuzzer)
if(JPEGXL_ENABLE_TOOLS)
  list(APPEND FUZZER_BINARIES
    color_encoding_fuzzer
    decode_basic_info_fuzzer
    cjxl_fuzzer
    icc_codec_fuzzer
    fields_fuzzer
    rans_fuzzer
    set_from_bytes_fuzzer
    transforms_fuzzer
  )
endif()
```

--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -307,17 +307,20 @@ endforeach()
 install(TARGETS ${TOOL_BINARIES} RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
 message(STATUS "Building tools: ${TOOL_BINARIES}")
 
-set(FUZZER_BINARIES
-  color_encoding_fuzzer
-  decode_basic_info_fuzzer
-  cjxl_fuzzer
-  djxl_fuzzer
-  icc_codec_fuzzer
-  fields_fuzzer
-  rans_fuzzer
-  set_from_bytes_fuzzer
-  transforms_fuzzer
-)
+# djxl_fuzzer builds even when not JPEGXL_ENABLE_TOOLS
+set(FUZZER_BINARIES djxl_fuzzer)
+if(JPEGXL_ENABLE_TOOLS)
+  list(APPEND FUZZER_BINARIES
+    color_encoding_fuzzer
+    decode_basic_info_fuzzer
+    cjxl_fuzzer
+    icc_codec_fuzzer
+    fields_fuzzer
+    rans_fuzzer
+    set_from_bytes_fuzzer
+    transforms_fuzzer
+  )
+endif()
 
 # Fuzzers.
 foreach(FUZZER IN LISTS FUZZER_BINARIES)

